auth/auth.go

```diff
@@ -1,21 +1,50 @@
 package auth
 
 import (
+	"io/ioutil"
+	"net/http"
+
+	jwt "github.com/dgrijalva/jwt-go"
+	"github.com/golang/glog"
 	"github.com/rancherio/host-api/app/common"
 	"github.com/rancherio/host-api/config"
-	jwt "github.com/dgrijalva/jwt-go"
-	"net/http"
 )
 
+var fileBytes []byte
+
 func Auth(rw http.ResponseWriter, req *http.Request) bool {
 	if !config.Config.Auth {
+		glog.Infoln("Host_API auth is disabled")
 		return true
 	}
+	getToken := req.URL.Query().Get("token")
 
-	token, err := jwt.ParseFromRequest(req, func(token *jwt.Token) (interface{}, error) {
-		return config.Config.Key, nil
+	if len(getToken) == 0 {
+		return false
+	}
+	glog.Infoln("Enabling Authencation for Host_API")
+
+	if len(fileBytes) == 0 {
+		file, err := ioutil.ReadFile(config.Config.Key)
+		if err != nil {
+			glog.Error("Error reading file")
+			return false
+		}
+		fileBytes = file
+	} else {
+		glog.Infoln("Reading cached content")
+	}
+	token, err := jwt.Parse(getToken, func(token *jwt.Token) (interface{}, error) {
+		return fileBytes, nil
 	})
 
+	if err == nil && token.Valid {
```

Comment:
I'm a bit confused to this log statement.  You log that the authentication is successful if there are no errors and the token is valid.  However, after you log this, you have additional checks on the hostUuid that could prove to be invalid and return a false.  The log message would then become misleading.